kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/MessageIdUtils.java

```diff
@@ -93,7 +95,7 @@ public void readEntryComplete(Entry entry, Object ctx) {
         return future;
     }
 
-    public static PositionImpl getPositionForOffset(ManagedLedger managedLedger, Long offset) {
+    public static PositionImpl getPositionForOffset(ManagedLedger managedLedger, Long offset) throws RuntimeException {
```

Comment:
Could you use a derived exception type instead of `RuntimeException`? Normally `RuntimeException` is an unchecked exception. I think it's better to create an exception that derives from `Exception` and throw it in `catch` clause.